File: VB/WebDocumentViewer_UserName/Services/CustomWebDocumentViewerOperationLogger.vb
Imports System
Imports System.Web
Imports DevExpress.XtraPrinting
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.Web.WebDocumentViewer

Namespace WebDocumentViewer_UserName.Services
    Friend Class CustomWebDocumentViewerOperationLogger
        Inherits WebDocumentViewerOperationLogger

        Public Overrides Function BuildStarting(ByVal reportId As String, ByVal report As XtraReport, ByVal buildProperties As ReportBuildProperties) As Action
            Dim httpContext = System.Web.HttpContext.Current
            Return Sub() GenerateBuildStatingAction(report, httpContext)
        End Function

        Private Shared Sub GenerateBuildStatingAction(ByVal report As XtraReport, ByVal httpContext As HttpContext)
            report.PrintingSystem.AddService(GetType(PageInfoDataProviderBase), New CustomPageInfoDataProvider(httpContext))
        End Sub
    End Class
End Namespace

File: CS/WebDocumentViewer_UserName/Services/CustomWebDocumentViewerOperationLogger.cs
using System;
using System.Web;
using DevExpress.XtraPrinting;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.Web.WebDocumentViewer;

namespace WebDocumentViewer_UserName.Services
{
    class CustomWebDocumentViewerOperationLogger : WebDocumentViewerOperationLogger
    {
        public override Action BuildStarting(string reportId, XtraReport report, ReportBuildProperties buildProperties)
        {
            var httpContext = HttpContext.Current;
            return () => GenerateBuildStatingAction(report, httpContext);
        }

        static void GenerateBuildStatingAction(XtraReport report, HttpContext httpContext)
        {
            report.PrintingSystem.AddService(typeof(PageInfoDataProviderBase), new CustomPageInfoDataProvider(httpContext));
        }
    }
}